pkg/controller/periodic/periodic.go

```diff
@@ -81,16 +82,23 @@ func (c *Controller) Run(stopCh <-chan struct{}, initialDelay time.Duration) {
 // Currently their is only 1 gatherer (clusterconfig) and no new gatherer is on the horizon.
 // Running the gatherers in parallel should be a future improvement when a new gatherer is introduced.
 func (c *Controller) Gather() {
+	interval := c.configurator.Config().Interval
+	threshold := status.GatherFailuresCountThreshold
 	for name := range c.gatherers {
-		start := time.Now()
-		err := c.runGatherer(name)
-		if err == nil {
-			klog.V(4).Infof("Periodic gather %s completed in %s", name, time.Since(start).Truncate(time.Millisecond))
-			c.statuses[name].UpdateStatus(controllerstatus.Summary{Healthy: true})
-			return
+		retry_count := 0
+		for retry_count < threshold {
+			time.Sleep(interval / time.Duration(threshold) * time.Duration(retry_count))
```

Comment:
So in case of an error we will wait 24 min right (second round 48 min) ? It seems too long....

Reply:
Maybe we could try to use https://godoc.org/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoff

Reply:
Yeah, I think the cleanest would be to try `X` times in between the intervals, this is trying to do that but math is hard and I overshot it a bit, while it should probably do the `X` retries in` interval-5m` or something.

The easiest implementation of this would be to just call `time.Sleep(interval / time.Duration(threshold))` so it will try after 24 minutes, does this 5 times.

Will try it with that https://godoc.org/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoff too, just for the fun of it :)